tests/unit/PureClarity/Api/Delta/Type/ProductTest.php

```diff
@@ -305,21 +295,17 @@ public function testSoftError()
         );
 
         $this->mockEndpoint();
-        $this->mockCurl($expectedBody, 400, 'Permission Denied');
-
-        $this->subject->addData(['SKU' => 'test123']);
-        $responses = $this->subject->send();
-
-        $this->assertEquals(1, count($responses));
+        $this->mockCurl($expectedBody, 200, 'An error response');
 
-        $this->assertEquals(
-            [
-                'status' => 400,
-                'body' => 'Permission Denied',
-                'error' => ''
-            ],
-            $responses[0]
-        );
+        try {
+            $this->subject->addData(['SKU' => 'test123']);
+            $this->subject->send();
+        } catch (Exception $e) {
+            $this->assertEquals(
+                'Error: HTTP 500 Response | Error Message: Some error | Body: ',
+                $e->getMessage()
+            );
+        }
```

Comment:
This assertion will never be fired. The response is successful, with a body containing "An error response" returned.

Ideally the structure here should be modified such that the assertion happens after the try catch, with the value asserted getting set within the try catch. This means we'll always fire the assertion